library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,836 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright: (c) 2019, F5 Networks Inc.
+# GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'certified'}
+
+DOCUMENTATION = r'''
+---
+module: bigip_gtm_dns_listener
+short_description: configures BIG-IP DNS system to answer TCP or UDP DNS requests
+description:
+  - defines one or more listener objects to control what protocols are available for the BIG-IP DNS system to process DNS requests
+  - BIG-IP DNS listeners allows to use TCP and UDP protocols
+version_added: "f5_modules 1.4"
+options:
+  name:
+    description:
+      - Specifies the name of the DNS Listener.
+    type: str
+    required: True
+  description:
+    description:
+      - Provides brief description for DNS Listener.
+    type: str
+    required: True
+  address:
+    description:
+      - specifies the IP address on which the system listens.
+    type: str
+  mask:
+    description:
+      - specifies the netmask for a network listener only.
+      - netmask clarifies whether the host bit is an actual zero or a wildcard representation.
+    default: ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff
+    required: True
+  pool:
+    description:
+      - specifies a default pool to which you want the listener to automatically direct traffic.
+    type: str
+  port:
+    description:
+      - specifies the port on which the listener listens for connections.
+      - Valid range of values is between C(0) and C(65535) inclusive.
+    type: int
+  source_port:
+    description:
+      - specifies whether the system preserves the source port of the connection.
+    type: str
+    default: preserve
+  translate_address:
+    description:
+      - enables or disables address translation for the listener.
+    type: str
+    default: enabled
+  translate_port:
+    description:
+      - enables or disables port translation.
+    type: str
+    default: enabled
+  vlans_enabled:
+    description:
+      - when C(yes), traffic will be accepted by this listener only on the VLANS specified in the vlans option.
+      - only C(yes) is allowed value
+    type: bool
+  vlans_disabled:
+    description:
+      - when C(yes), traffic will not be accepted by this listener only on the VLANS specified in the vlans option.
+      - only C(yes) is allowed value
+    type: bool
+  vlans:
+    description:
+      - specifies a list of VLANs on which traffic is either disabled or enabled.
+      - enabling/disabling decision is based on C(vlans_disabled) and C(vlans_enabled) settings.
+      - removing/reseting VLANs required providing empyt list C([]) as parameter value
+    type: list
+    elements: str
```

Comment:
can this endpoint support vlans enabled and disabled at the same time, just like VS? if yes then redo this completely

Reply:
In the API doc, it states the following: "Specifies a list of VLANs on which traffic is either disabled or enabled, based on the vlans-disabled and vlans-enabled settings." 
https://clouddocs.f5.com/api/icontrol-rest/APIRef_tm_gtm_listener.html

Testing shows that vlans_enabled and vlans_disabled are mutually exclusive. BIGIP prefers vlans_disabled in a case when both flags provided.

Also, looking at VirtualServer module, it appears that in a case when both provided, VS module takes only disabled vlans: 
https://github.com/F5Networks/f5-ansible/blob/devel/library/modules/bigip_virtual_server.py#L2090 


In my implementation, there are three fields, such as: 
   - vlans_enabled  - flag which indicates what to do; mutually exclusive with vlans_disabled
   - vlans_disabled - flag which indicates what to do; mutually exclusive with vlans_enabled. 
   - vlans - list of vlans. Setting to empty list "[]" will make all option 

Let me know if you have  any questions/concerns